src/machine_charm.py

```diff
@@ -147,14 +147,13 @@ class GrafanaAgentMachineCharm(GrafanaAgentCharm):
 
     service_name = "grafana-agent.grafana-agent"
 
-    # Pairs of (incoming, outgoing) relation names. If any 'incoming' is joined without a matching
-    # 'outgoing', the charm will block. Without an outgoing relation we may incur data loss.
+    # Pairs of (incoming, [outgoing]) relation names. If any 'incoming' is joined
+    # without a matching 'outgoing', the charm will block. Without an outgoing relation
+    # we may incur data loss.
     mandatory_relation_pairs = [
-        ("cos-agent", "send-remote-write"),
-        ("cos-agent", "logging-consumer"),
-        ("cos-agent", "grafana-dashboards-provider"),
-        ("juju-info", "send-remote-write"),
-        ("juju-info", "logging-consumer"),
+        ("cos-agent", ["send-remote-write", "logging-consumer", "grafana-cloud-config"]),
+        ("cos-agent", ["grafana-dashboards-provider"]),
+        ("juju-info", ["send-remote-write", "logging-consumer", "grafana-cloud-config"]),
```

Comment:
I would argue we also shouldn't require the dashboard relation. Seems fine to leave that optional. If they want it and forgot, they will notice when they check the dashboards and no historical data will have been lost.

Reply:
:+1:

Reply:
done